gtsam/hybrid/DCMixtureFactor.h

```diff
@@ -116,15 +137,13 @@ class DCMixtureFactor : public DCFactor {
     // object from `other`
     const DCMixtureFactor& f(static_cast<const DCMixtureFactor&>(other));
 
-    // Ensure that this DCMixtureFactor and `f` have the same number of
-    // component factors in `factors_`.
-    if (factors_.size() != f.factors_.size()) return false;
-
-    // If the number of factors is the same, we compare them individually (they
-    // should be in the same order!). If any fail to match, return false.
-    for (size_t i = 0; i < factors_.size(); i++) {
-      if (!factors_[i].equals(f.factors_[i])) return false;
-    }
+    // Ensure that this DCMixtureFactor and `f` have the same `factors_`.
+    auto comparator = [](const boost::shared_ptr<NonlinearFactorType>& a,
```

Comment:
nice! *but* check in Testable. There is actually a shared pointer version of Equals called equals_star. Maybe it's just:
```
    if (!factors_.equals(f.factors_, 1e-9, equals_star<NonlinearFactorType>)) return false;
```

Reply:
I though about that, but the comparator requires 3 arguments `(a, b, tol)` whereas `equals_star` takes two.
`tol` is passed in as the constructor argument to `equals_star`.